starknet-client/src/client/contract_interaction.rs

```diff
@@ -0,0 +1,266 @@
+//! Defines `StarknetClient` helpers that allow for interacting with the `darkpool` contract
+
+use circuit_types::merkle::MerkleRoot;
+use circuit_types::native_helpers::compute_wallet_commitment_from_private;
+use circuit_types::traits::BaseType;
+use circuit_types::traits::CircuitCommitmentType;
+use circuit_types::wallet::Nullifier;
+use circuit_types::wallet::WalletShareStateCommitment;
+use circuit_types::SizedWalletShare;
+use common::types::proof_bundles::OrderValidityProofBundle;
+use common::types::proof_bundles::ValidMatchMpcBundle;
+use common::types::proof_bundles::ValidSettleBundle;
+use common::types::proof_bundles::ValidWalletCreateBundle;
+use common::types::proof_bundles::ValidWalletUpdateBundle;
+use mpc_stark::algebra::scalar::Scalar;
+use renegade_crypto::fields::scalar_to_starknet_felt;
+use starknet::accounts::Call;
+use starknet::core::types::FieldElement as StarknetFieldElement;
+use starknet::core::types::FunctionCall;
+
+use crate::types::CalldataSerializable;
+use crate::types::ExternalTransfer;
+use crate::types::MatchPayload;
+use crate::GET_PUBLIC_BLINDER_TRANSACTION;
+use crate::MATCH_SELECTOR;
+use crate::NEW_WALLET_SELECTOR;
+use crate::NULLIFIER_USED_SELECTOR;
+use crate::UPDATE_WALLET_SELECTOR;
+use crate::{error::StarknetClientError, MERKLE_ROOT_IN_HISTORY_SELECTOR};
+
+use super::StarknetClient;
+use super::TransactionHash;
+
+impl StarknetClient {
+    // --- Getters ---
+
+    /// Check whether the given Merkle root is valid
+    pub async fn check_merkle_root_valid(
+        &self,
+        root: MerkleRoot,
+    ) -> Result<bool, StarknetClientError> {
+        // TODO: Implement BigUint on the contract
+        let reduced_root = scalar_to_starknet_felt(&root);
```

Comment:
nit: not sure if this should be called `reduced` here (and elsewhere), since there's no "reduction" from scalar to felt (as the field is smaller)

Reply:
oops yes forgot to change this after the curve refactor